### PR TITLE
WebSocket (step 4) - NAS experiment

### PR DESCRIPTION
**Pull request by liuzhe-lz**
### Description ###

Migrate NAS experiment to WebSocket

Following steps:

  - Support multiple experiments in one process
  - Migrate old UTs and remove unused files

### Checklist ###
  - [x] test case
  - ~~doc~~

### CODE REVIEW

File: nni/retiarii/experiment/pytorch.py [hunk 1]
```diff
@@ -354,9 +353,11 @@ def start(self, port: int = 8080, debug: bool = False) -> None:
             log_dir = Path.home() / f'nni-experiments/{self.id}/log'
         nni.runtime.log.start_experiment_log(self.id, log_dir, debug)
 
-        self._proc, self._pipe = launcher.start_experiment_retiarii(self.id, self.config, port, debug)
+        ws_url = f'ws://localhost:{port}/tuner'
+        self._proc = launcher.start_experiment('create', self.id, self.config, port, debug,  # type: ignore
+                                               RunMode.Background, None, ws_url, ['retiarii'])
         assert self._proc is not None
-        assert self._pipe is not None
+        connect_websocket(ws_url)
```
Comment (matluster) [inline nni/retiarii/experiment/pytorch.py:360]: When do you disconnect?

Comment (cruiseliu) [reply nni/retiarii/experiment/pytorch.py:360]: On exit 😂

File: nni/experiment/launcher.py [hunk 2]
```diff
@@ -129,12 +149,12 @@ def start_experiment(action, exp_id, config, port, debug, run_mode, url_prefix):
...
     return proc
 
-def _start_rest_server(nni_manager_args, run_mode) -> Popen:
+def _start_rest_server(nni_manager_args: NniManagerArgs, run_mode: RunMode) -> Popen:
     import nni_node
     node_dir = Path(nni_node.__path__[0])  # type: ignore
     node = str(node_dir / ('node.exe' if sys.platform == 'win32' else 'node'))
     main_js = str(node_dir / 'main.js')
-    cmd = [node, '--max-old-space-size=4096', main_js]
+    cmd = [node, '--max-old-space-size=4096', '--trace-uncaught', main_js]
```
Comment (QuanluZhang) [inline nni/experiment/launcher.py:157]: what is `--trace-uncaught` used for?

**Review by QuanluZhang — APPROVED** [review]

File: nni/retiarii/experiment/pytorch.py (hunk 1 above)
Comment (matluster) [reply nni/retiarii/experiment/pytorch.py:360]: What if I stop the experiment in the middle the process. When I do something afterwards (e.g., export), would the websocket client still ping-pong?

File: nni/experiment/launcher.py (hunk 2 above)
Comment (cruiseliu) [reply nni/experiment/launcher.py:157]: Print stack trace when an uncaught exception is thrown in async function. For better debugability.

**Review by ultmaster — APPROVED** [review]

**Pull request merged**